### PR TITLE
Add skill book variants, 1/?

### PR DESCRIPTION
**Pull request by GuardianDll**
#### Summary
Content "Some skill books got variations"
#### Purpose of change
Did you know we have only 2 books to train unarmed combat? only 3 books to train archery? Not anymore
#### Describe the solution
Added variations to some books in the game - most are real, but some are made:
archery, barter, bashing, chemistry, computers, cutting, melee and unarmed in this pr
#### Testing
The game works pretty well, but book variations has pretty intreresting issue - if you read one book from the variation, you will know the containement of all another books of this variation
#### Additional context
Maybe ill make an organizing issue, as it is a big project, and i want to make it fairly easy to review and proof-read
image

### CODE REVIEW

File: data/json/items/book/archery.json [hunk 1]
```diff
@@ -22,6 +22,29 @@
...
+      {
+        "id": "book_archery",
+        "name": { "str": "Lessons for the Novice Bowhunter", "str_pl": "copies of Lessons for the Novice Bowhunter" },
+        "description": "This hefty paperback book contains all the information needed for novice archers to get started hunting with a variety of bows and crossbows.",
+        "weight": 1
+      },
+      {
+        "id": "book_archery_1",
+        "name": { "str": "Archery Anatomy", "str_pl": "copies of Archery Anatomy" },
+        "description": "Book, that highlight archery techniques from the point of view of the relationship between the anatomy of the human body and the anatomy of the bow.",
```
Comment (Maleclypse) [inline data/json/items/book/archery.json:35]:
```suggestion
        "description": "A book that highlights archery techniques from the point of view of the relationship between the anatomy of the human body and the anatomy of the bow.",
```

File: data/json/items/book/archery.json [hunk 2]
```diff
@@ -31,6 +54,29 @@
...
+      {
+        "id": "manual_archery",
+        "name": { "str": "Zen and the Art of Archery", "str_pl": "copies of Zen and the Art of Archery" },
+        "description": "This massive book contains a wealth of vital information for the novice archer.",
+        "weight": 1
+      },
+      {
+        "id": "manual_archery_1",
+        "name": {
+          "str": "Beginner s Guide to Traditional Archery Sorrells",
```
Comment (Maleclypse) [inline data/json/items/book/archery.json:67]:
```suggestion
          "str": "Beginner's Guide to Traditional Archery Sorrells",
```

File: data/json/items/book/barter.json [hunk 3]
```diff
@@ -24,6 +24,26 @@
...
+      {
+        "id": "manual_business",
+        "name": { "str": "How to Succeed in Business", "str_pl": "copies of How to Succeed in Business" },
+        "description": "Useful if you want to get a good deal when purchasing goods.",
+        "weight": 1
+      },
+      {
+        "id": "manual_business_1",
+        "name": { "str": "How To Day Trade For A Living", "str_pl": "copies of How To Day Trade For A Living" },
+        "description": "Author describe the fundamentals of day trading, explain how day trading is different from other styles of trading and investment, and elaborate on important trading strategies that many traders use every day.",
```
Comment (Maleclypse) [inline data/json/items/book/barter.json:37]:
```suggestion
        "description": "Author describes the fundamentals of day trading, explain how day trading is different from other styles of trading and investment, and elaborate on important trading strategies that many traders use every day.",
```

File: data/json/items/book/barter.json [hunk 4]
```diff
@@ -24,6 +24,26 @@
...
+        "weight": 1
+      },
+      {
+        "id": "manual_business_1",
+        "name": { "str": "Trading Psychology For Dummies", "str_pl": "copies of Trading Psychology For Dummies" },
+        "description": "Helps you develop a mindset to trade on a rapidly changes stock market.  Not much useful in current moment, when the main stock you have is a stock of your gun.",
```
Comment (Maleclypse) [inline data/json/items/book/barter.json:43]:
```suggestion
        "description": "Helps you develop a mindset to trade on a rapidly changing stock market.  Not much useful in current moment, when the main stock you have is a stock of your gun, or maybe a soup.",
```

File: data/json/items/book/barter.json [hunk 5]
```diff
@@ -45,6 +65,27 @@
...
+      {
+        "id": "textbook_business",
+        "name": { "str": "Advanced Economics", "str_pl": "copies of Advanced Economics" },
+        "description": "A college textbook on economics.",
+        "weight": 1
+      },
+      {
+        "id": "textbook_business_1",
+        "name": { "str": "Peer to Peer Bartering", "str_pl": "copies of Peer to Peer Bartering" },
+        "description": "In this work author adopt bartering incentive pattern as an attractive foundation for a simple and robust form of exchange to re allocate resources.",
```
Comment (Maleclypse) [inline data/json/items/book/barter.json:78]:
```suggestion
        "description": "In this work author adopt bartering incentive pattern as an attractive foundation for a simple and robust form of exchange to reallocate resources.",
```

File: data/json/items/book/barter.json [hunk 6]
```diff
@@ -45,6 +65,27 @@
...
+        "price": 60000,
+        "weight": 1
+      },
+      {
+        "id": "textbook_business_2",
+        "name": { "str": "Trading Basics", "str_pl": "copies of Trading Basics" },
+        "description": "Comprehensive coverage of the most major trading styles.",
```
Comment (Maleclypse) [inline data/json/items/book/barter.json:85]:
```suggestion
        "description": "Comprehensive coverage of most major trading styles.",
```

File: data/json/items/book/barter.json (hunk 5 above)
Comment (Maleclypse) [inline data/json/items/book/barter.json:78]:
```suggestion
        "description": "In this work author adopt bartering incentive pattern as an attractive foundation for a simple and robust form of exchange to reallocate resources.",
```

File: data/json/items/book/chemistry.json [hunk 7]
```diff
@@ -289,6 +388,29 @@
...
+          "str": "Handbook of Essential Oils: Science, Technology, and Applications",
+          "str_pl": "copies of Handbook of Essential Oils: Science, Technology, and Applications"
+        },
+        "description": "This book presents the development, use and marketing of essential oils, with wide range of topic, sources, production and analysis.",
+        "weight": 1
+      },
+      {
+        "id": "textbook_extraction_2",
+        "name": { "str": "Essential Oil Distillation", "str_pl": "copies of Essential Oil Distillation" },
+        "description": "This book provides a detailed guide to the process of distilling essential oils, including information on equipment and process of extracting them from plant materials. It also includes tips for selecting and storing essential oils, as well as using the byproducts of manufacturing.",
```
Comment (GuardianDll) [inline data/json/items/book/chemistry.json:410]:
```suggestion
        "description": "This book provides a detailed guide to the process of distilling essential oils, including information on equipment and process of extracting them from plant materials.  It also includes tips for selecting and storing essential oils, as well as using the byproducts of manufacturing.",
```

File: data/json/items/book/cutting.json [hunk 8]
```diff
@@ -5,6 +5,41 @@
     "category": "manuals",
     "name": { "str": "All About Swords", "str_pl": "issues of All About Swords" },
     "description": "An interesting magazine that contains information about swords and sword fighting techniques from all across the world.",
+    "variants": [
+      {
+        "id": "mag_cutting",
+        "name": { "str": "All About Swords", "str_pl": "issues of All About Swords" },
+        "description": "An interesting magazine that contains information about swords and sword fighting techniques from all across the world..",
```
Comment (GuardianDll) [inline data/json/items/book/cutting.json:12]:
```suggestion
        "description": "An interesting magazine that contains information about swords and sword fighting techniques from all across the world.",
```

File: data/json/items/book/melee.json [hunk 9]
```diff
@@ -3,8 +3,43 @@
     "id": "mag_melee",
     "type": "BOOK",
     "category": "manuals",
-    "name": { "str": "CQB Monthly", "str_pl": "CQB Monthlies" },
+    "name": { "str": "CQB Monthly", "str_pl": "issues of CQB Monthly" },
     "description": "An in-depth look at various styles of close quarters fighting.  There's an amusing essay about dirty tricks in the front section.",
+    "variants": [
+      {
+        "id": "mag_melee",
+        "name": { "str": "CQB Monthly", "str_pl": "copies of CQB Monthly" },
```
Comment (GuardianDll) [inline data/json/items/book/melee.json:11]:
```suggestion
        "name": { "str": "CQB Monthly", "str_pl": "issues of CQB Monthly" },
```

Comment (GuardianDll) [inline data/json/items/book/melee.json:11]:
```suggestion
        "name": { "str": "CQB Monthly", "str_pl": "issues of CQB Monthly" },
```

File: data/json/items/book/melee.json [hunk 10]
```diff
@@ -3,8 +3,43 @@
...
+      {
+        "id": "mag_melee_1",
+        "name": { "str": "Savage Street Fighting", "str_pl": "copies of Savage Street Fighting" },
+        "description": "Barbaric guide how to win a fight in worst possible scenarios - mostly about how to retreat from such fight, but still has some information in the cases you have no another way.",
+        "weight": 1
+      },
+      {
+        "id": "mag_melee_2",
+        "name": { "str": "Maximum Damage", "str_pl": "copies of Maximum Damage" },
+        "description": "The book teach user the method to predict enemies attacks, with nice explanation of how to reach on each of them - both defense and counterattacks.",
```
Comment (GuardianDll) [inline data/json/items/book/melee.json:24]:
```suggestion
        "description": "The book teach user the method to predict enemies attacks, with nice explanation of how to react on each of them - both defense and counterattacks.",
```

File: data/json/items/book/unarmed.json [hunk 11]
```diff
@@ -25,6 +60,50 @@
...
+        "name": {
+          "str": "Killer Instinct: Unarmed Combat for Street Survival",
+          "str_pl": "copies of Killer Instinct: Unarmed Combat for Street Survival"
+        },
+        "description": "Shabby book about possible street dangers author met in his long life, ways to deal with them, and skill set, that such situations require.",
+        "weight": 1
+      },
+      {
+        "id": "manual_brawl_5",
+        "name": { "str": "The Bigger They Are, The Harder They Fall", "str_pl": "The Bigger They Are, The Harder They Fall" },
```
Comment (GuardianDll) [inline data/json/items/book/unarmed.json:102]:
```suggestion
        "name": { "str": "The Bigger They Are, The Harder They Fall", "str_pl": "copies of The Bigger They Are, The Harder They Fall" },
```

Comment (GuardianDll) [inline data/json/items/book/unarmed.json:102]:
```suggestion
        "name": { "str": "The Bigger They Are, The Harder They Fall", "str_pl": "copies of The Bigger They Are, The Harder They Fall" },
```

File: data/json/items/book/chemistry.json [hunk 12]
```diff
@@ -373,6 +575,72 @@
...
+        "description": "The book contain information on different steps in producing biofuels in different scales - from home-made work in your backyard to large manufacturing - with different issues you may met in the process, and ways to resolve them.",
+        "weight": 1
+      },
+      {
+        "id": "textbook_biodiesel_5",
+        "name": {
+          "str": "Biodiesel Basics and Beyond: A Comprehensive Guide to Production and Use for the Home and Farm",
+          "str_pl": "copies of Biodiesel Basics and Beyond: A Comprehensive Guide to Production and Use for the Home and Farm"
+        },
+        "description": "Big reference book about using biodiesel in homemade scale, with lot of how-tos and original thoughts about processes.",
```
Comment (GuardianDll) [inline data/json/items/book/chemistry.json:625]:
```suggestion
        "description": "Big reference book about using biodiesel in homemade scale, with lot of how-to's and original thoughts about processes.",
```

File: tools/spell_checker/dictionary.txt [hunk 13]
```diff
@@ -2729,6 +2738,7 @@ topickup
 toroids
 torosaurus
 torvosaurus
+tos
```
Comment (GuardianDll) [inline tools/spell_checker/dictionary.txt:2741]:
```suggestion
```

File: data/json/items/book/chemistry.json [hunk 14]
```diff
@@ -373,6 +575,72 @@
...
+        "intelligence": 8,
+        "weight": 1
+      },
+      {
+        "id": "textbook_biodiesel_3",
+        "name": {
+          "str": "Biodiesel: a Realistic Fuel Alternative for Diesel Engines",
+          "str_pl": "copies of Biodiesel: a Realistic Fuel Alternative for Diesel Engines"
+        },
+        "description": "Small book, that offer insights into plant-based fuels, with detailed analysis of biodiesel, and possible scenarios how it can be implemented in .",
```
Comment (TheShadowFerret) [inline data/json/items/book/chemistry.json:607]:
```suggestion
        "description": "A small book offering insights into plant-based fuels, with detailed analysis of biodiesel, and possible scenarios for their implementation.",
```

File: data/json/items/book/chemistry.json [hunk 15]
```diff
@@ -373,6 +575,72 @@
...
+        "description": "The book contain information on different steps in producing biofuels in different scales - from home-made work in your backyard to large manufacturing - with different issues you may met in the process, and ways to resolve them.",
+        "weight": 1
+      },
+      {
+        "id": "textbook_biodiesel_5",
+        "name": {
+          "str": "Biodiesel Basics and Beyond: A Comprehensive Guide to Production and Use for the Home and Farm",
+          "str_pl": "copies of Biodiesel Basics and Beyond: A Comprehensive Guide to Production and Use for the Home and Farm"
+        },
+        "description": "Big reference book about using biodiesel in homemade scale, with lot of how-to's and original thoughts about processes.",
```
Comment (TheShadowFerret) [inline data/json/items/book/chemistry.json:625]:
```suggestion
        "description": "A comprehensive guide by William Kemp to produce biodiesel at small scale while avoiding common failures of fuel quality and byproduct waste management, with accuracies of data and process backed by government or university verification.",
```

File: data/json/items/book/chemistry.json [hunk 16]
```diff
@@ -373,6 +575,72 @@
...
+          "str": "Handbook of Biofuels Production: Process and Technologies",
+          "str_pl": "copies of Handbook of Biofuels Production: Process and Technologies"
+        },
+        "description": "Book about different types of biofuels, issues in their production and using, and different processes and technologies that are used in their manufacturing.",
+        "weight": 1
+      },
+      {
+        "id": "textbook_biodiesel_7",
+        "name": { "str": "Bean Diesel", "str_pl": "copies of Bean Diesel" },
+        "description": "The original cover of this book was replaced with white sheet of paper with hand made \"Bean Diesel\" label and simplified, draw by pen, image of Vin Diesel.  Except of this, it is pretty mundane handbook about biofuel production.",
```
Comment (TheShadowFerret) [inline data/json/items/book/chemistry.json:640]:
```suggestion
        "description": "The original cover of this book was replaced with a white sheet of paper with a hand-made \"Bean Diesel\" label and simplified, draw by pen, image of Vin Diesel.  Excluding this, it is a pretty mundane handbook about biofuel production.",
```

File: data/json/items/book/computer.json [hunk 17]
```diff
@@ -5,6 +5,34 @@
...
+      {
+        "id": "SICP_2",
+        "name": { "str": "The Art of Computer Programming, Vol. 1", "str_pl": "copies of The Art of Computer Programming, Vol. 1" },
+        "description": "Huge reference  manual about fundamental algorithms, with lot of examples and math.",
+        "fun": -2,
+        "weight": 1
+      },
+      {
+        "id": "SICP_3",
+        "name": { "str": "The Art of Computer Programming, Vol. 3", "str_pl": "copies of The Art of Computer Programming, Vol. 3" },
```
Comment (TheShadowFerret) [inline data/json/items/book/computer.json:30]: Where did volume 2 go 🤔

**Review by TheShadowFerret — COMMENTED** [review]
~~Was this PR meant to get people to track down which books you used~~

1K lines is a lot to proofread :(

File: data/json/items/book/computer.json (hunk 17 above)
Comment (GuardianDll) [reply data/json/items/book/computer.json:30]: Volume 2 ~~was lost~~ is about seminumerical algorithms, and there is no information about what it is that i can use in description
There is also volume 4, that i also ignored, about combinatorial algorithms

File: data/json/items/book/chemistry.json (hunk 14 above)
Comment (GuardianDll) [reply data/json/items/book/chemistry.json:607]: Huh, seems i forgot to finish this one :derpy:

**Comment by GuardianDll** [comment]
I'll try to make it smaller in the future

File: data/json/items/book/bashing.json [hunk 18]
```diff
@@ -5,6 +5,41 @@
...
+        "description": "A baseball magazine that focuses on batting tips.  There are lots of colorful, full-page photos of skilled athletes demonstrating proper form and technique.",
+        "weight": 1
+      },
+      {
+        "id": "mag_bashing_1",
+        "name": {
+          "str": "Bludgeons and Broken Bones: A Reference Guide",
+          "str_pl": "copies of Bludgeons and Broken Bones: A Reference Guide"
+        },
+        "description": "Writed with hard prison slang, this book says a lot about how to use maximum power from your muscles and weight of your weapon.",
```
Comment (GuardianDll) [inline data/json/items/book/bashing.json:21]:
```suggestion
        "description": "Written with hard prison slang, this book says a lot about how to use maximum power from your muscles and weight of your weapon.",
```

**Pull request merged**